Fix: Prevent TypeError on Executor Shutdown During Garbage Collection

ihb2032 opened the pull request:
## Purpose

This PR fixes a `TypeError` that occurs during the garbage collection of the `ExecutorBase` object upon program exit.

When the vLLM engine shuts down, a race condition can occur where dependent objects (like `collective_rpc` in `UniprocExecutor`) are destroyed before `ExecutorBase.__del__` calls the `shutdown()` method. This results in a `TypeError: 'NoneType' object is not callable` and prints an "Exception ignored" message to the console.

This PR prevents the exception by wrapping the `shutdown()` call in a `try...except TypeError` block. This allows for a graceful and silent exit, even if shutdown dependencies are no longer available during the final stages of interpreter shutdown.

## Test Plan

1.  Run any vLLM example script to completion.
2.  Observe the console output when the script finishes to ensure no traceback is printed.
3.  A simple test command:
    ```bash
python qwen_riscv.py
    ```

This was specifically tested and verified on the following environment where the issue was initially observed:

  * **Architecture**: `sg2044riscv64`
  * **OS**: `EulixOS 3.0`
  * **PyTorch**: `2.8.0`

## Test Result

The fix successfully eliminates the traceback on exit.

**Before this change:**

The following traceback was printed to the console upon script completion:

```
Exception ignored in: <function ExecutorBase.__del__ at 0x7fff9e8eeb60>
Traceback (most recent call last):
  File "/AI/hebo/vllm/vllm/executor/executor_base.py", line 237, in __del__
  File "/AI/hebo/vllm/vllm/executor/executor_base.py", line 234, in shutdown
  File "/AI/hebo/vllm/vllm/executor/uniproc_executor.py", line 58, in collective_rpc
TypeError: 'NoneType' object is not callable
```

**After this change:**

The script completes successfully without printing any `Exception ignored` traceback. The exit is clean.

Commit:
fix(executor): Handle TypeError during graceful shutdown in ExecutorBase.__del__

During the Python interpreter shutdown process, the order of object garbage collection is not guaranteed. This can lead to a race condition where the `shutdown()` method is called in `ExecutorBase.__del__`, but some dependent objects (like `collective_rpc` in `UniprocExecutor`) have already been destroyed and set to `None`.
This results in a `TypeError: 'NoneType' object is not callable`, printing an ignored exception traceback to the console, which can be confusing.
This commit wraps the `self.shutdown()` call within a `try...except` block to catch `TypeError`. This ensures a clean and silent exit without raising an exception if the shutdown process is interrupted by garbage collection.

Platform Context:
- Arch: sg2044riscv64
- OS: EulixOS 3.0
- GCC: 15.1
- PyTorch: 2.8.0
- Python: 3.11.6

    Signed-off-by: lyd1992 <liuyudong@iscas.ac.cn>
    Signed-off-by: ihb2032 <1355790728@qq.com>

ihb2032 committed:
Merge branch 'main' into fix/executor-shutdown-typeerror

Commit:
fix(executor): Gracefully handle exceptions in __del__ during shutdown

Catches both TypeError and AttributeError in ExecutorBase.__del__ to prevent crashes during interpreter shutdown when objects may be in an unpredictable state.

    Signed-off-by: lyd1992 <liuyudong@iscas.ac.cn>
    Signed-off-by: ihb2032 <1355790728@qq.com>

Commit:
Merge branch 'fix/executor-shutdown-typeerror' of https://github.com/ihb2032/vllm into fix/executor-shutdown-typeerror

ihb2032 committed:
Merge branch 'main' into fix/executor-shutdown-typeerror

chaunceyjiang reviewed (commented):
https://github.com/vllm-project/vllm/pull/24621

Can you test it again with the latest version of vLLM? This issue should already be fixed.

Commit:
refactor(executor): Use contextlib.suppress for exception handling

Replace the `try...except...pass` block in `vllm/executor/executor_base.py` with `contextlib.suppress`.
This change addresses the ruff linting error SIM105, making the intent to ignore specific exceptions more explicit and fixing the failing pre-commit CI check.

Signed-off-by: lyd1992 <liuyudong@iscas.ac.cn>
Signed-off-by: ihb2032 <1355790728@qq.com>

Commit:
Merge branch 'fix/executor-shutdown-typeerror' of https://github.com/ihb2032/vllm into fix/executor-shutdown-typeerror
